### PR TITLE
address thread safety issues

### PR DESCRIPTION
**Pull request by tomerd**
motivation: recent changes to TSC exposed several thread safety issues

changes:
* make build description and graph caches thread safe
* simplify repository manager in-memory cache and make it thread-safe
* adjust tests

### CODE REVIEW

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Comment by tomerd** [comment]
cc @pcbeard

File: Sources/SourceControl/RepositoryManager.swift [hunk 1]
```diff
@@ -400,26 +397,15 @@ extension RepositoryManager {
...
-        /// The status of the repository.
-        fileprivate var status: Status
-
-        /// Lock to protect  the operations like updating the state
-        /// of the handle and fetching the repositories from its remote.
-        private let statusLock = Lock()
-
         /// Create a handle.
-        fileprivate init(manager: RepositoryManager, repository: RepositorySpecifier, subpath: RelativePath, status: Status = .uninitialized) {
+        fileprivate init(manager: RepositoryManager, repository: RepositorySpecifier, subpath: RelativePath/*, status: Status = .uninitialized*/) {
```
Comment (abertelrud) [inline Sources/SourceControl/RepositoryManager.swift:401]:
```suggestion
        fileprivate init(manager: RepositoryManager, repository: RepositorySpecifier, subpath: RelativePath) {
```

**Review by abertelrud — APPROVED** [review]

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Review by neonichu — APPROVED** [review]

**Pull request merged**